compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt

```diff
@@ -111,14 +110,15 @@ internal class RenderingUIView(
             onAttachedToWindow?.invoke()
             _isReadyToShowContent.value = true
         }
+        updateMetalLayerSize()
     }
 
     override fun layoutSubviews() {
         super.layoutSubviews()
         updateMetalLayerSize()
     }
 
-    internal fun updateMetalLayerSize() {
+    private fun updateMetalLayerSize() {
```

Comment:
Since this function is also called when the window is attached, it's possible that frame could be zero or contentScaleFactor is wrong. We either need to instate an invariant that it's not the case and put the check/assert, or to return early, if the order of availability of valid `frame` and `window?.screen?.scale` is assumed to be uncertain.

Reply:
Done